harness/determined/common/experimental/user.py

```diff
@@ -5,6 +5,28 @@
 
 
 class User:
+    """
+    A User object represents an individual account on a Determined installation.
+    It can be obtained from ``client.list_users`` or ``client.get_user_by_name()``.
+
+    Attributes:
+        session: HTTP request session.
+        user_id: Unique ID for the user.
+        admin: (Mutable) Admin status of the user.
+        remote: (Mutable) Checkpoint associated with this model version.
+        agent_uid: (Mutable) Unix user ID on the agent the task runs on.
```

Comment:
Why should IDs be mutable?

Reply:
well, they shouldn't be, and i agree the documentation here is a bit confusing. it's mutable in the sense that it might be null when initialized. not sure how to best express that, maybe `(Optional[int])`?

Reply:
I think it's ok to spell it out with this many words in the docstring. Either set of symbols is fine alongside.

Reply:
updated docstrings to include an `Optional[Type]` description